### PR TITLE
Add experimental support for repeatable ArgGroup positional parameters (#1027)

### LINKED ISSUE #1027: Support repeatable ArgGroups with positional parameters
See https://stackoverflow.com/a/61606437/1446916

We would like to be able to parse user input like this (repeating pairs of `name grade`):

```
grades Abby 4.0 Billy 3.5 Caily 3.5 Danny 4.0
```

Ideally using strongly-typed positional parameters. 
A natural way to model this would be with an argument group, like this:

```java
// unfortunately this does not work...
static class StudentGrade {
    @Parameters(index = "0") String name;
    @Parameters(index = "1") BigDecimal grade;
}

@ArgGroup(exclusive = false, multiplicity = "1..*")
List<StudentGrade> gradeList;
```

However, this gives the following error:

```
Unmatched arguments from index 2: 'Billy', '3.5', 'Caily', '3.5', 'Danny', '4.0'
Usage: grades (<name> <grade>)... [-hV]
      <name>
      <grade>
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
```

### PR DESCRIPTION
**Pull request by NewbieOrange**
This is a very simple draft to fix #1027, and does not break any previous tests.

Could anyone check whether any behaviour breaks with this?

### CODE REVIEW

**Comment by NewbieOrange** [comment]
With the following test
```java
    @Test
    public void testIssue1027RepeatingPositionalParamsWithMaxMultiplicity() {
        class Issue1027 {
            @ArgGroup(exclusive = false, multiplicity = "1..3")
            List<StudentGrade> gradeList;
        }

        Issue1027 bean = new Issue1027();
        new CommandLine(bean).parseArgs("Abby 4.0 Billy 3.5 Caily 3.5 Danny 4.0".split(" "));

        assertEquals(3, bean.gradeList.size());
        assertEquals(new StudentGrade("Abby", "4.0"), bean.gradeList.get(0));
        assertEquals(new StudentGrade("Billy", "3.5"), bean.gradeList.get(1));
        assertEquals(new StudentGrade("Caily", "3.5"), bean.gradeList.get(2));
    }
```
Currently with this PR, picocli throws
```
picocli.CommandLine$MaxValuesExceededException: Error: expected only one match but got (<name> <grade>) [<name> <grade>] [<name> <grade>]={params[0]=Abby params[1]=4.0 params[0]=Billy params[1]=3.5 params[0]=Caily params[1]=3.5} and (<name> <grade>) [<name> <grade>] [<name> <grade>]={params[0]=Danny params[1]=4.0}
```
The exception is as expected, but the message is very user-unfriendly: "only one match" and "got !@#$".

**Comment by NewbieOrange** [comment]
The min multiplicity test gives reasonable exception message though
```
picocli.CommandLine$MissingParameterException: Error: Group: (<name> <grade>) must be specified 4 times but was matched 3 times
```

File: src/main/java/picocli/CommandLine.java [hunk 1]
```diff
@@ -12453,7 +12453,12 @@ private void processPositionalParameter(Collection<ArgSpec> required, Set<ArgSpe
                 int localPosition = getPosition(positionalParam);
                 if (positionalParam.group() != null) { // does the positionalParam's index range contain the current position in the currently matching group
                     GroupMatchContainer groupMatchContainer = parseResultBuilder.groupMatchContainer.findOrCreateMatchingGroup(positionalParam, commandSpec.commandLine());
-                    if (!indexRange.contains(localPosition) || (groupMatchContainer != null && groupMatchContainer.lastMatch().hasMatchedValueAtPosition(positionalParam, localPosition))) {
+                    int argGroupPosition = localPosition;
+                    if (groupMatchContainer != null) {
```
Comment (NewbieOrange) [inline src/main/java/picocli/CommandLine.java:12457]: This condition might be unnecessary

**Comment by remkop** [comment]
Overall looks good.
Thanks for another deep dive in a difficult area!
Still going over a few things, will probably merge tomorrow.

**Comment by remkop** [comment]
Merged into master.
Thank you for the pull request!

**Pull request merged**